api/src/neo4j/queries/reaction.js

```diff
@@ -59,6 +68,10 @@ RETURN apoc.map.mergeList(COLLECT(value.data)) as reaction
 `;
 
   const reaction = await querySingleResult(statement);
+  if (reaction.geneRule) {
+    reaction.geneRule = formatGeneRule(reaction);
+  }
+
```

Comment:
Conceptually, I would move all necessary checks in the gene rule formatting.

```suggestion
  reaction.geneRule = formatGeneRule(reaction);
```

Reply:
Good suggestion! I had some problems with this when I tried it before but it seems fine now actually.